packages/@sanity/portable-text-editor/src/utils/values.ts

```diff
@@ -67,40 +82,47 @@ export function fromSlateValue(
 ): PortableTextBlock[] {
   if (value && Array.isArray(value)) {
     return value.map((block) => {
-      const isPortableText = block && block._type === textBlockType
-      if (isPortableText && Element.isElement(block)) {
-        let hasInlines = false
-        const children = block.children.map((child) => {
-          const {_type} = child
-          if (_type !== 'span' && typeof child.value === 'object') {
-            hasInlines = true
-            // eslint-disable-next-line @typescript-eslint/no-unused-vars
-            const {value: v, _key: k, _type: t, __inline: _i, children: _c, ...rest} = child
+      if (Element.isElement(block)) {
+        if (block._type === textBlockType) {
+          let hasInlines = false
+          const children = block.children.map((child) => {
+            const {_type} = child
+            if (
+              Element.isElement(child) &&
+              'value' in child &&
+              _type !== 'span' &&
+              typeof child.value === 'object'
+            ) {
+              hasInlines = true
+              // eslint-disable-next-line @typescript-eslint/no-unused-vars
+              const {value: v, _key: k, _type: t, __inline: _i, children: _c, ...rest} = child
+              return keepObjectEquality(
+                {...rest, ...v, _key: k as string, _type: t as string},
+                keyMap
+              )
+            }
+            return child
+          })
+          if (typeof block._key === 'string' && typeof block._type === 'string') {
+            if (!hasInlines) {
+              // Original object
+              return block
+            }
             return keepObjectEquality(
-              {_key: k as string, _type: t as string, ...rest, ...v},
+              {...block, children, _key: block._key, _type: block._type},
               keyMap
             )
           }
-          return child as PortableTextChild
-        })
-        if (typeof block._key === 'string' && typeof block._type === 'string') {
-          if (!hasInlines) {
-            // Original object
-            return (block as unknown) as PortableTextBlock
-          }
-          return keepObjectEquality(
-            {_key: block._key, _type: block._type, ...block, children},
-            keyMap
-          ) as PortableTextBlock
+          throw new Error('Not a valid block type')
         }
-        throw new Error('Not a valid block type')
+        const {_key, _type} = block
+        const blockValue = 'value' in block && block.value
+        return keepObjectEquality(
+          {_key, _type, ...(typeof blockValue === 'object' ? blockValue : {})},
+          keyMap
+        ) as PortableTextBlock
```

Comment:
Is the `as PortableTextBlock` type casting required?

Reply:
Yes, as it's a conversion util we have to manually assign the types here.